pkg/param/param.go

```diff
@@ -90,12 +91,14 @@ type CredentialType string
 
 const (
 	CredentialTypeKeyPair CredentialType = "keyPair"
+	CredentialTypeAWS     CredentialType = "aws"
```

Comment:
Should we make this `CredentialTypeSecret` too? In case we want something similar for Azure.
@SupriyaKasten

Reply:
My thinking was, at this point, we should materialize the type information. For @SupriyaKasten's case, we can also add `CredentialTypeAzure`.